packages/app/src/components/DataTable.tsx

```diff
@@ -1,90 +1,256 @@
-import { Button, Dialog } from "@mui/material"
-import { useRecoilValue } from "recoil";
-import { dataSelector } from "../atoms/dataAtom";
-import { Row, getRows, isRowAggregate } from "@visdesignlab/upset2-core";
-import { useContext, useMemo } from "react";
-import { ProvenanceContext } from "./Root";
+import { Box, Button } from "@mui/material"
+import { AccessibleDataEntry, CoreUpsetData } from "@visdesignlab/upset2-core";
+import { useMemo } from "react";
 import { DataGrid, GridColDef } from '@mui/x-data-grid';
+import { getAccessibleData } from "@visdesignlab/upset2-react";
+import DownloadIcon from '@mui/icons-material/Download';
 
-const getRowData = (row: Row) => {
-    return {id: row.id, elementName: `${(isRowAggregate(row)) ? "Aggregate: " : ""}${row.elementName}`, cardinality: row.size}
+const getRowData = (row: AccessibleDataEntry) => {
+    return {id: row.id, elementName: `${(row.type === "Aggregate") ? "Aggregate: " : ""}${row.elementName}`, size: row.size}
 }
 
-const getAggRows = (row: Row) => {
+const getAggRows = (row: AccessibleDataEntry) => {
     const retVal: ReturnType<typeof getRowData>[] = [];
+    if (row.items === undefined) return retVal;
 
-    Object.values(row.items.values).forEach((r: Row) => {
+    Object.values(row.items).forEach((r: AccessibleDataEntry) => {
         retVal.push(getRowData(r));
 
-        if (isRowAggregate(r)) {
+        if (r.type === "Aggregate") {
+            console.log("is agg")
             retVal.push(...getAggRows(r));
         }
     });
 
     return retVal;
 }
 
-export const DataTable = (props: {close: () => void}) => {
-    const { provenance } = useContext(ProvenanceContext);
+const downloadCSS = {
+    margin: "4px",
+    height: "40%",
+}
+
+const headerCSS = {
+    display: "flex",
+    justifyContent: "space-between",
+    alignItems: "center",
+    margin: "2px"
+}
+
+function downloadElementsAsCSV(items: any[], columns: string[], name: string) {
+    if (items.length < 1 || columns.length < 1) return;
+  
+    console.group(name);
+    console.log(columns);
+    console.table(items.filter((_, idx) => idx < 10));
```

Comment:
Why so much console action? Is this an accessibility affordance, or just for debugging?

Reply:
This is a relic from debugging. Removed it